File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {

    List<TbBrand> findAll() throws Exception;

    PageResult<TbBrand> findByPage(TbBrand brand, int pageNum, int pageSize) throws Exception;

    TbBrand findById(Long id) throws Exception;

    void save(TbBrand brand) throws Exception;

    void delete(Long[] ids) throws Exception;

    List<Map<String, Object>> getOptionList() throws Exception;
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/plugins/angularjs/service/brandService.js
app.service('brandService', function ($http) {

    this.findByPage = function (page, size, searchEntity) {
        return $http.post('../brand/findByPage.do?page=' + page + '&size=' + size, searchEntity);
    };

    this.findById = function (id) {
        return $http.get('../brand/findById.do?id=' + id);
    };

    this.save = function (entity) {
        return $http.post('../brand/save.do', entity);
    };

    this.del = function (ids) {
        return $http.get('../brand/del.do?ids=' + ids);
    };

    this.getOptionList = function () {
        return $http.get('../brand/getOptionList.do');
    };
});